### PR TITLE
Edited the .script/build.ps1 power shell script so that the lines wit…

### PR DESCRIPTION
**Pull request by jmartinez989**
…h a call to Robocopy.exe do not have the /DCOPY:DA argument on them anymore.

Also added code to make Faction units armor and weapons buildable. Also made bullpups have the same mobility and concealment benefits as smgs. Also adjusted damage values for faction unit weapons to be higher from mag tier and up so that they can scale with the scaling enemies.

### CODE REVIEW

File: LongWarOfTheChosen/Config/XComEngine.ini [hunk 1]
```diff
@@ -15,6 +15,7 @@
 +ModEditPackages=LW_AlienPack_Integrated
 +ModEditPackages=LW_Toolbox_Integrated
 +ModEditPackages=LW_Overhaul
++ModEditPackages=WotCWeaponsAndArmor
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComEngine.ini]: I'll put the commend here. Could you rename this? Probably to `LW_WeaponsAndArmor`. Just for consistency. Plus I think I'd like to refactor other weapon and armour code into this package at some point.

Also, when you do use WOTC, please use all caps for consistency throughout the project. Plus, it's much easier on my fingers to type all caps or all lower case :smile:

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini [hunk 2]
```diff
@@ -1768,3 +1768,210 @@ SPARK_NOVA_SELF_BASEDAMAGE= (Damage=0, Spread = 0, PlusOne = 0, Crit = 0, Pierce
 FERALMEC_M1_PROXY_WPN_BASEDAMAGE=(Damage=2, Spread=0, PlusOne=25, Crit=1, Pierce=0, Shred=0, Tag="", DamageType="Projectile_Conventional")
 
 SECTOPOD_MARKOV_PROXY_WPN_BASEDAMAGE=(Damage=3, Spread = 0, PlusOne = 25, Crit = 1, Pierce = 0, Shred = 1, Tag = "", DamageType="Projectile_Conventional")
+
+[WotCWeaponsAndArmor.X2Item_Weapons]
+;Bukllpup stats
+BULLPUP_CONVENTIONAL_BASEDAMAGE=(Damage=3, Spread=0, PlusOne=50, Crit=1, Pierce=0, Shred=0, Tag="", DamageType="Projectile_Conventional")
+BULLPUP_MAGNETIC_BASEDAMAGE=(Damage=6, Spread=0, PlusOne=50, Crit=2, Pierce=0, Shred=0, Tag="", DamageType="Projectile_MagXCom")
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: This is one more damage than the SMG, whereas the other tiers are the same as SMG. Intentional? Also, is it intentional to have no spread on this weapons? I guess that's what the PlusOne is for. It makes the range effectively 5-6.

Hmmm...not sure what to do about the faction weapons in the long term. Feel like they need to be more distinctive.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini [hunk 3]
```diff
@@ -1768,3 +1768,210 @@ SPARK_NOVA_SELF_BASEDAMAGE= (Damage=0, Spread = 0, PlusOne = 0, Crit = 0, Pierce
...
+BULLPUP_BEAM_BASEDAMAGE=(Damage=8, Spread=0, PlusOne=50, Crit=4, Pierce=0, Shred=0, Tag="", Rupture = 1, DamageType="Projectile_BeamXCom")
+
+BULLPUP_CONVENTIONAL_AIM = 0
+BULLPUP_CONVENTIONAL_CRITCHANCE = 0
+BULLPUP_CONVENTIONAL_ICLIPSIZE = 3
+BULLPUP_CONVENTIONAL_ISOUNDRANGE = 30
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Just want to point out that this is slightly higher than SMG. I'm happy with that.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini [hunk 4]
```diff
@@ -1768,3 +1768,210 @@ SPARK_NOVA_SELF_BASEDAMAGE= (Damage=0, Spread = 0, PlusOne = 0, Crit = 0, Pierce
...
+
+SIDEARM_BEAM_AIM = 0
+SIDEARM_BEAM_CRITCHANCE = 0
+SIDEARM_BEAM_ICLIPSIZE = 99
+SIDEARM_BEAM_ISOUNDRANGE = 15
+SIDEARM_BEAM_IENVIRONMENTDAMAGE = 1
+
+;Weapon Range Tables
+VEKTOR_MAGNETIC_RANGE[0] = 0
+VEKTOR_MAGNETIC_RANGE[1] = 26
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Why 0 for point blank? Seems weird. Is this based off of vanilla numbers?

Comment (pledbrook) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Or is zero not point blank? It means "on top of"?

File: LongWarOfTheChosen/Config/XComWeaponAbilities.ini [hunk 5]
```diff
@@ -0,0 +1,12 @@
+;A configuration File
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComWeaponAbilities.ini]: Why are these not in `XComGameData_WeaponData.ini`? That's where the SMG mobility bonus stuff is. I think it you make sense to keep them in the same config file.

File: LongWarOfTheChosen/LongWarOfTheChosen.x2proj [hunk 6]
```diff
@@ -1,1112 +1,1116 @@
...
-    <Content Include="Src\LW_Overhaul\Classes\X2TargetingMethod_LWBlasterLauncher.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\X2TargetingMethod_LWEvacZone.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\X2TargetingMethod_LWRocketLauncher.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_Effect_Bombard.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_Effect_Failsafe.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_Effect_ImmediateMultiTarget.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_Effect_JavelinRockets.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_Effect_TransferMecToOutpost.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_LWAlienActivity.uc" />
-    <Content Include="Src\LW_Overhaul\Classes\XComGameState_LWAl
```
Comment (pledbrook) [inline LongWarOfTheChosen/LongWarOfTheChosen.x2proj]: Hmmm...I think this may be a line ending issue. Can you restore the original line endings so that it doesn't look like the whole file has changed?

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc [hunk 7]
```diff
@@ -0,0 +1,123 @@
+//---------------------------------------------------------------------------------------
+//  FILE:    X2Ability_SMGAbilities.uc
+//  AUTHOR:  Amineri (Long War Studios)
+//  PURPOSE: Adds stat modifier abilities for all tech tiers of SMG
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: I would update the text here if you plan to use it :smile:

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc [hunk 8]
```diff
@@ -0,0 +1,123 @@
+//---------------------------------------------------------------------------------------
+//  FILE:    X2Ability_SMGAbilities.uc
+//  AUTHOR:  Amineri (Long War Studios)
+//  PURPOSE: Adds stat modifier abilities for all tech tiers of SMG
+//           
+//---------------------------------------------------------------------------------------
+class X2Ability_WeaponAbilities extends X2Ability config(WeaponAbilities);
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: Could you please rename to something more specific? Perhaps `X2Ability_BullpupAbilities` or `X2Ability_FactionWeaponAbilities`?

Comment (pledbrook) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: And yes, I'd prefer it if the config loaded from `GameData_WeaponData`.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2DownloadableContentInfo_WotCWeaponsAndArmor.uc [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+//  FILE:   XComDownloadableContentInfo_WaroftheChosenRebalance.uc                                    
+//           
+//	Use the X2DownloadableContentInfo class to specify unique mod behavior when the 
+//  player creates a new campaign or loads a saved game.
+//  
+//---------------------------------------------------------------------------------------
+//  Copyright (c) 2016 Firaxis Games, Inc. All rights reserved.
+//---------------------------------------------------------------------------------------
+
+class X2DownloadableContentInfo_WotCWeaponsAndArmor extends X2DownloadableContentInfo;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2DownloadableContentInfo_WotCWeaponsAndArmor.uc]: Is this class needed? I don't think it is if it doesn't do anything. Not sure though.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc [hunk 10]
```diff
@@ -0,0 +1,265 @@
+// This is an Unreal Script
+class X2Item_Armors extends X2Item;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: How about `X2Item_FactionArmors`?

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+	Template.PointsToComplete = 0;
+	Template.Abilities.AddItem('MediumPlatedArmorStats');
+	Template.ArmorTechCat = 'plated';
+	Template.ArmorCat = 'templar';
+	Template.Tier = 1;
+	Template.AkAudioSoldierArmorSwitch = 'Predator';
+	Template.EquipNarrative = "X2NarrativeMoments.Strategy.CIN_ArmorIntro_PlatedMedium";
+	Template.EquipSound = "StrategyUI_Armor_Equip_Plated";
+
+    Template.Requirements.RequiredTechs.AddItem('PlatedArmor');
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: Bonus points if you manage to make indent consistent, i.e. tabs instead of spaces. Note that the LW2 code base is a mess when it comes to whitespace, so this would fit right in :smile: Would prefer consistency though.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+	Schematics.AddItem(CreateTemplate_WristBlade_Magnetic_Schematic());
+	Schematics.AddItem(CreateTemplate_WristBlade_Beam_Schematic());
+
+	Schematics.AddItem(CreateTemplate_ShardGauntlet_Magnetic_Schematic());
+	Schematics.AddItem(CreateTemplate_ShardGauntlet_Beam_Schematic());
+
+	Schematics.AddItem(CreateTemplate_Sidearm_Magnetic_Schematic());
+	Schematics.AddItem(CreateTemplate_Sidearm_Beam_Schematic());
+
+	return Schematics;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc]: If you want to remove the schematics from the game, just remove them from [this list](https://github.com/long-war-2/lwotc/blob/master/LongWarOfTheChosen/Config/XComLW_Overhaul.ini#L326) in `XComGameLW_Overhaul.ini`.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc [hunk 13]
```diff
@@ -0,0 +1,923 @@
+// This is an Unreal Script
+class X2Item_Weapons extends X2Item config(GameData_WeaponData);
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc]: To be consistent with the LW2 way of doing things, these weapons should probably be added to the [`ItemTable` array](https://github.com/long-war-2/lwotc/blob/master/LongWarOfTheChosen/Config/XComLW_Overhaul.ini#L448) in `XComLW_Overhaul.ini`.

File: buildlog.txt [hunk 14]
```diff
@@ -0,0 +1,2820 @@
+c:\Vagrant\boxes\shared\lwotc\
```
Comment (pledbrook) [inline buildlog.txt]: As I think you're aware, this shouldn't be in the commit. Same with the Alien Base plots (no idea why they're different).

**Comment by pledbrook** [comment]
Also meant to mention that if there is an associated issue, please put the issue number (with #) in the commit message. I put them in parentheses in the first line, but it doesn't have to be done like that.

File: LongWarOfTheChosen/Config/XComEngine.ini (hunk 1 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComEngine.ini]: OK I will rename this. I will rename it to LWOTC_WeaponsAndArmor. Does that work?

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 2 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: So as far as the faction weapons go, they do not have a spread. That is intended and the only thing I tweaked there is the base damage. I made them higher so that they can deal with the scaling health enemies as the elite enemies have quite a bit more hp than in vanilla.

I am going to have have bullpups sit at the lowest damage, vektor rifles sit around the middle and have templars have the highest as they require more risk.

Not sure about what the scondaries. I might just keep them as they are tbh. Will have to have a secondary discussion about that.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 3 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Yeah I wanted them to have just a bit more ammo than smg since they can actually hold more.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 4 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: No this is just an oopsie on my part. Didn't see it was at zero. I'll fix this.

File: LongWarOfTheChosen/Config/XComWeaponAbilities.ini (hunk 5 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComWeaponAbilities.ini]: Ok I will move the bullpup config information to the XComGameData_WeaponData.ini file for consistency.

File: LongWarOfTheChosen/LongWarOfTheChosen.x2proj (hunk 6 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/LongWarOfTheChosen.x2proj]: I never touched that file so I do not know what happened here. What is it exactly I need to change back?

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc (hunk 7 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: D'oh yeah I meant to give him credit for that and put myself as the author :P. I'll make sure to get that taken care of.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc (hunk 8 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: Ok I will rename it to X2Ability_FactionWeaponAbilities.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2DownloadableContentInfo_WotCWeaponsAndArmor.uc (hunk 9 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2DownloadableContentInfo_WotCWeaponsAndArmor.uc]: I don't think I need this class right now but I am going to leave it for the time being as I might need to use it for some last minute processing later.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc (hunk 10 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: Ok I will rename to X2Item_FactionArmors.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc (hunk 11 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: Arrrgghh I am a spaces guy myself :P. I'll make sure to get the indenting fixed here.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc (hunk 12 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc]: Ok cool, was not aware that was there. I'll remove them from that file then and get rid of the schematics class then.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc (hunk 13 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc]: Ahh ok so if I add them to that file I can edit the costs there and not have to worry about setting the costs at template creation then?

File: buildlog.txt (hunk 14 above)
Comment (jmartinez989) [reply buildlog.txt]: Yeah my bad on the .txt file. Forgot to remove that. As with the base plot, I think that those files are not part of the .gitignore file in the content directory. I can change that if needed.

File: LongWarOfTheChosen/Config/XComEngine.ini (hunk 1 above)
Comment (pledbrook) [reply LongWarOfTheChosen/Config/XComEngine.ini]: I still prefer just the LW_ prefix for consistency unless you're planning to release as a separate mod? I'm ok with that, just want to check.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 3 above)
Comment (pledbrook) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Sorry, was referring to sound range.

File: LongWarOfTheChosen/Config/XComEngine.ini (hunk 1 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComEngine.ini]: ok sure, I'll use LW_instead.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 3 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Ahh, yeah that too. Forgot I had set that higher.

File: LongWarOfTheChosen/LongWarOfTheChosen.x2proj (hunk 6 above)
Comment (pledbrook) [reply LongWarOfTheChosen/LongWarOfTheChosen.x2proj]: The build automatically verifies that all the files under `LongWarOfTheChosen` are listed in the ModBuddy project file. Since you added some new files, the project file was updated.

What are you using to edit the files? Most text editors give an indication of what lime endings are in use and allow you to swap between them. Standard ones are DOS, Unix and Mac.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc (hunk 11 above)
Comment (pledbrook) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: Trust me, I prefer spaces too, but the convention for highlander and most of LW2 is tabs. When in Rome...

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc (hunk 13 above)
Comment (pledbrook) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc]: Correct. These values are applies to existing templates in the `LWTemplateMods` class.

File: buildlog.txt (hunk 14 above)
Comment (pledbrook) [reply buildlog.txt]: The plots are explicitly excluded from gitignore because they are stored in the repository. I just don't know why you have "modified" versions. I assume by accident.

File: LongWarOfTheChosen/LongWarOfTheChosen.x2proj (hunk 6 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/LongWarOfTheChosen.x2proj]: Oh I was using sublime 3 editor.

File: LongWarOfTheChosen/Config/XComEngine.ini (hunk 1 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComEngine.ini]: This has been changed to LW_WeaponsAndArmor.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 4 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Fixed this. Now Vektor at point blank is 30.

File: LongWarOfTheChosen/Config/XComWeaponAbilities.ini (hunk 5 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComWeaponAbilities.ini]: Configs for this are now in GameData_WeaponData.

File: LongWarOfTheChosen/LongWarOfTheChosen.x2proj (hunk 6 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/LongWarOfTheChosen.x2proj]: Ok so now this file seems to have proper line endings. The diffs only show the lines that were added instead of the whole file.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc (hunk 7 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: Fixed. Now the line reads me as the author and gives credit to Amineri.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc (hunk 8 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Ability_WeaponAbilities.uc]: This has been renamed.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc (hunk 10 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: This has been renamed.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc (hunk 11 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Armors.uc]: All of the files should now be using tabs instead of spaces for indentation.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc (hunk 13 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Weapons.uc]: These are now added to the itemtable array in LW_Overhaul.ini

File: buildlog.txt (hunk 14 above)
Comment (jmartinez989) [reply buildlog.txt]: Removed this file from repo.

File: LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc (hunk 12 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/WotCWeaponsAndArmor/Classes/X2Item_Schematics.uc]: Removed the schematics from LW_Overhaul.ini so now the schematic class is not needed to hide them.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini [hunk 15]
```diff
@@ -1978,12 +1978,12 @@ TEMPLAR_PISTOL_RANGE[25] = -100
 
 [LW_WeaponsAndArmor.X2Ability_FactionWeaponAbilities]
 // ***** Mobility bonuses for Bullpups
-BULLPUP_CONVENTIONAL_MOBILITY_BONUS = 3
-BULLPUP_MAGNETIC_MOBILITY_BONUS = 3
-BULLPUP_BEAM_MOBILITY_BONUS = 3
+BULLPUP_CONVENTIONAL_MOBILITY_BONUS = 2
+BULLPUP_MAGNETIC_MOBILITY_BONUS = 2
+BULLPUP_BEAM_MOBILITY_BONUS = 2
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Not quite a cleanup of unneeded files :wink:

File: LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc [hunk 16]
```diff
@@ -1,5 +1,5 @@
 // This is an Unreal Script
-class X2Item_Weapons extends X2Item config(GameData_WeaponData);
+class X2Item_FactionWeapons extends X2Item config(GameData_WeaponData);
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc]: I'm confused. Is this file needed still, since the weapon configuration is in the `ItemTable` array and the templates already exist?

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini [hunk 17]
```diff
@@ -1868,7 +1868,7 @@ SIDEARM_BEAM_ISOUNDRANGE = 15
 SIDEARM_BEAM_IENVIRONMENTDAMAGE = 1
 
 ;Weapon Range Tables
-VEKTOR_MAGNETIC_RANGE[0] = 0
+VEKTOR_MAGNETIC_RANGE[0] = 30
```
Comment (pledbrook) [inline LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: You updated Vektor rifle, but not Bullpup or Autopistol.

File: LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc (hunk 16 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc]: hmmm well this is actually a good point. I think it will still be needed unless I can somehow apply the weapon ability to the bullpups via the item table. If I can then I should be able to delete the class. If not then I will need to keep it so that I can apply the ability to the bullpups.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 17 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: LOL Dangit. ok I'll fix those.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 15 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: Yeah that got included in part of that commit. I forgot to mention that change as part of that commit. Unless you meant something else.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 17 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: This has been changed.

File: LongWarOfTheChosen/Config/XComGameData_WeaponData.ini (hunk 15 above)
Comment (pledbrook) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: It's better to separate independent, unrelated changes into different commits. I can probably clean this up when I merge.

Comment (jmartinez989) [reply LongWarOfTheChosen/Config/XComGameData_WeaponData.ini]: ahh ok gotcha. I'll make sure to make unrelated changes separate commits.

File: LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc (hunk 16 above)
Comment (jmartinez989) [reply LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_FactionWeapons.uc]: As per our conversation on discord I am resolving this conversation as we are choosing to keep the templates.

**Comment by pledbrook** [comment]
Superseded by #119
